doc/conf.py

```diff
@@ -260,6 +261,13 @@
 # Output file base name for HTML help builder.
 htmlhelp_basename = 'zephyrdoc'
 
+
+# Custom added feature to allow redirecting old URLs
+#
+# list of tuples (old_url, new_url) for pages to redirect
+# (URLs should be relative to document root, only)
+html_redirect_pages = [('contribute/contribute_guidelines', 'contribute/index'),]
```

Comment:
I'm not sure how we'll maintain this redirect list (both to add new redirects and remove redirects that aren't needed any more).  

A different approach is to manually create an html redirect file and add it to the folder where the .rst file was (Sphinx will ignore them), and then post-process the rst folders to copy any .html file to the corresponding folder in the generated html tree (see https://tech.signavio.com/2017/managing-sphinx-redirects).  This way you can drop these html redirect file in the source tree when you delete or move the .rst file vs. remembering to edit the conf.py file.

Reply:
> I'm not sure how we'll maintain this redirect list (both to add new redirects and remove redirects that aren't needed any more).

update the tuple. Alternatively we could maintain those in a CSV type file. This is for main entry points, not for every file that has been moved.



> A different approach is to manually create an html redirect file and add it to the folder where the .rst file was (Sphinx will ignore them), and then post-process the rst folders (after the Sphinx build) to copy any .html file to the corresponding folder in the generated html tree (see https://tech.signavio.com/2017/managing-sphinx-redirects). (Need to skip the _templates folder.) This way you can drop these html redirect file in the source tree when you delete or move the .rst file vs. remembering to edit the conf.py file.

I do not see this being easier or easier to maintain as the above approach.